src/workerd/api/actor-state.c++

```diff
@@ -939,6 +939,39 @@ kj::Maybe<kj::Date> DurableObjectState::getWebSocketAutoResponseTimestamp(jsg::R
   return ws->getAutoResponseTimestamp();
 }
 
+void DurableObjectState::setHibernatableWebSocketEventTimeout(jsg::Optional<uint32_t> timeout) {
+  auto& a = KJ_REQUIRE_NONNULL(IoContext::current().getActor());
+
+  // Setting a timeout = 0ms or an empty value will unset any currently set event timeout.
+  // If there's no hibernation manager instantiated, we can skip the event timeout unsetting.
+  if (timeout == kj::none || KJ_REQUIRE_NONNULL(timeout) == 0) {
+    KJ_IF_SOME(hibernationManager, a.getHibernationManager()) {
+      hibernationManager.setEventTimeout(kj::none);
+    }
+    return;
+  }
+
+  auto t = KJ_REQUIRE_NONNULL(timeout); // set timeout
+
+  // We want to limit the duration of an event to a maximum of 7 days (604800 * 1000 millis).
+  JSG_REQUIRE(t <= 604800 * 1000, Error, "Event timeout should not exceed 604800000 ms.");
+  KJ_IF_SOME(hibernationManager, a.getHibernationManager()) {
+    hibernationManager.setEventTimeout(t);
+  } else {
+    a.setHibernationManager(kj::refcounted<HibernationManagerImpl>(
+        a.getLoopback(), KJ_REQUIRE_NONNULL(a.getHibernationEventType(), t)));
```

Comment:
Is the intent here to create a new hibernation manager and set its event timeout to t?  Because I think passing t as the second parameter to KJ_REQUIRE_NONNULL will just have the effect of stringifying t as the message to display when KJ_REQUIRE_NONNULL fails.

Reply:
Nice catch!